sql/core/src/main/scala/org/apache/spark/sql/execution/window/WindowExec.scala

```diff
@@ -87,7 +88,8 @@ case class WindowExec(
     windowExpression: Seq[NamedExpression],
     partitionSpec: Seq[Expression],
     orderSpec: Seq[SortOrder],
-    child: SparkPlan)
+    child: SparkPlan,
+    groupLimitInfo: Option[(Int, Expression)] = None)
```

Comment:
I think it's overkill and very risky to make invasive changes to a fundamental physical operator like `WindownExec`. I like https://github.com/apache/spark/pull/34367 more which adds a new physical node. Can you elaborate on why is this better than https://github.com/apache/spark/pull/34367 ?

Reply:
I think it is OK to add a physical node, but the amount of code is a little large, and the filtering and reduction of data occur a little late.